lib/src/db/db.dart

```diff
@@ -399,12 +398,12 @@ Future _commitHelper(DatastoreDB db,
   });
 }
 
-Future<List<Model>> _lookupHelper(DatastoreDB db, List<Key> keys,
+Future<List<T>> _lookupHelper<T extends Model>(DatastoreDB db, List<Key> keys,
     {ds.Transaction datastoreTransaction}) {
   var entityKeys = keys.map(db.modelDB.toDatastoreKey).toList();
   return db.datastore
       .lookup(entityKeys, transaction: datastoreTransaction)
       .then((List<ds.Entity> entities) {
-    return entities.map(db.modelDB.fromDatastoreEntity).toList();
+    return entities.map(db.modelDB.fromDatastoreEntity).cast<T>().toList();
```

Comment:
Since `fromDatastoreEntity` is now `fromDatastoreEntity<T>` we should be able to avoid the extra `.cast<T>()` here. Something like this should work:

```dart
return entities.map<T>(db.modelDB.fromDatastoreEntity).toList()
```

This uses a new language feature in Dart 2.0 which is a "partial instantiation". Implicitly it should do this:


```dart
var closure = db.modelDB.fromDatastoreEntity;
var closureWithType = closure<T>;
return entities.map<T>(closure).toList()
```

Disclaimer: Have not tested it.

Reply:
Nice! Added the change.